### PR TITLE
Ftr/api map config

### PR DESCRIPTION
**Pull request by nobnisai**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,6 +5,6 @@ gem 'immutable-struct'
 gem "rgeo"
 gem "rgeo-geojson"
 gem "pg", (ENV['GEM_PG_VERSION'] ? "~> #{ENV['GEM_PG_VERSION']}" : "~> 1.2.2") # make sure we use a version compatible with AR
-gem "rgeo-activerecord", (ENV['GEM_RGEO_ACTIVERECORD_VERSION'] ? "~> #{ENV['GEM_RGEO_ACTIVERECORD_VERSION']}" : "~> 7.0.1") # same as above
-gem 'activerecord-postgis-adapter', (ENV['GEM_ACTIVERECORD_POSTGIS_ADAPTER_VERSION'] ? "~> #{ENV['GEM_ACTIVERECORD_POSTGIS_ADAPTER_VERSION']}" : "~> 7.1.1") # same as above
+gem "rgeo-activerecord", (ENV['GEM_RGEO_ACTIVERECORD_VERSION'] ? "~> #{ENV['GEM_RGEO_ACTIVERECORD_VERSION']}" : "~>  6.2.2") # same as above
+gem 'activerecord-postgis-adapter', (ENV['GEM_ACTIVERECORD_POSTGIS_ADAPTER_VERSION'] ? "~> #{ENV['GEM_ACTIVERECORD_POSTGIS_ADAPTER_VERSION']}" : "~> 5.2.3") # same as above
```
Comment (dkastl) [inline Gemfile:9]: Any reason to downgrade versions?

Comment (nobnisai) [reply Gemfile:9]: Oh, there're some errors in my local so I downgrade the version. will make the change in the next commit.

File: Gemfile [hunk 2]
```diff
@@ -5,6 +5,6 @@ gem 'immutable-struct'
 gem "rgeo"
 gem "rgeo-geojson"
 gem "pg", (ENV['GEM_PG_VERSION'] ? "~> #{ENV['GEM_PG_VERSION']}" : "~> 1.2.2") # make sure we use a version compatible with AR
-gem "rgeo-activerecord", (ENV['GEM_RGEO_ACTIVERECORD_VERSION'] ? "~> #{ENV['GEM_RGEO_ACTIVERECORD_VERSION']}" : "~> 7.0.1") # same as above
+gem "rgeo-activerecord", (ENV['GEM_RGEO_ACTIVERECORD_VERSION'] ? "~> #{ENV['GEM_RGEO_ACTIVERECORD_VERSION']}" : "~>  7.0.1") # same as above
```
Comment (sanak) [inline Gemfile:8]: @nobnisai There is still diff with this space.

File: app/controllers/gtt_configuration_controller.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                defaultMapSetting: {
+                    centerLng: Setting.plugin_redmine_gtt['default_map_center_longitude'],
+                    centerLat: Setting.plugin_redmine_gtt['default_map_center_latitude']
+                },
+                geometrySetting: {
+                    geometryTypes: Setting.plugin_redmine_gtt['editable_geometry_types_on_issue_map'],
+                    GeoJsonUpload: Setting.plugin_redmine_gtt['enable_geojson_upload_on_issue_map'],
+                },
+                geocoderSetting: {
+                    enableGeocodingOnMap: Setting.plugin_redmine_gtt['enable_geocoding_on_map'],
```
Comment (sanak) [inline app/controllers/gtt_configuration_controller.rb:66]: @nobnisai (CC: @dkastl)  
Could you set the value like as follows to pass boolean value ?
```rb
enableGeocodingOnMap: (Setting.plugin_redmine_gtt['enable_geocoding_on_map'] == 'true')
```

File: app/controllers/gtt_configuration_controller.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            gttDefaultSetting: {
+                defaultTrackerIcon: default_tracker_icon,
+                defaultStatusColor: default_status_color,
+                defaultMapSetting: {
+                    centerLng: Setting.plugin_redmine_gtt['default_map_center_longitude'],
+                    centerLat: Setting.plugin_redmine_gtt['default_map_center_latitude']
+                },
+                geometrySetting: {
+                    geometryTypes: Setting.plugin_redmine_gtt['editable_geometry_types_on_issue_map'],
+                    GeoJsonUpload: Setting.plugin_redmine_gtt['enable_geojson_upload_on_issue_map'],
```
Comment (sanak) [inline app/controllers/gtt_configuration_controller.rb:63]: @nobnisai (CC: @dkastl)  
Could you set the value like as follows to pass boolean value ?
```rb
GeoJsonUpload: (Setting.plugin_redmine_gtt['enable_geojson_upload_on_issue_map'] == 'true')
```

File: app/views/projects/show.api.rsb [hunk 5]
```diff
@@ -14,6 +14,10 @@ api.project do
     api.geojson ""
   end
 
+  if @project.gtt_tile_sources.present?
+    api.gtt_tile_sources @project.gtt_tile_sources
+  end
+
```
Comment (sanak) [inline app/views/projects/show.api.rsb:20]: @nobnisai @dkastl (CC: @mopinfish, @smellman)
About exposing tile sources information, let me confirm this more from other point of view. 🙇‍♂️

File: config/routes.rb [hunk 6]
```diff
@@ -5,3 +5,7 @@
 put 'projects/:id/settings/gtt',
   to: 'projects#update_gtt_configuration',
   as: :update_gtt_configuration
+
+scope 'gtt' do
+  get 'settings', to: 'gtt_configuration#default_setting_configuration', as: :default_setting_configuration
+end
```
Comment (sanak) [inline config/routes.rb:11]: @nobnisai 
This is a slight thing, but ending with a new line may be better for compare diff.

File: app/controllers/gtt_configuration_controller.rb [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+    def find_optional_project_and_authorize
+        if params[:project_id]
+            @project = Project.find params[:project_id]
+            authorize
+        else
+            authorize_global
+        end
+    end
+end
```
Comment (sanak) [inline app/controllers/gtt_configuration_controller.rb:74]: @nobnisai
Also, ending with a new line may be better for compare diff.

**Review by dkastl — APPROVED** [review]

**Pull request merged**